benchmarks/Microsoft.AspNetCore.SignalR.Microbenchmarks/HubProtocolBenchmark.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using BenchmarkDotNet.Attributes;
+using Microsoft.AspNetCore.SignalR.Internal;
+using Microsoft.AspNetCore.SignalR.Internal.Encoders;
+using Microsoft.AspNetCore.SignalR.Internal.Protocol;
+
+namespace Microsoft.AspNetCore.SignalR.Microbenchmarks
+{
+    [ParameterizedJobConfig(typeof(CoreConfig))]
+    public class HubProtocolBenchmark
+    {
+        private HubProtocolReaderWriter _hubProtocolReaderWriter;
+        private byte[] _binaryInput;
+        private TestBinder _binder;
+        private HubMessage _hubMessage;
+
+        [Params(Message.Small, Message.Medium, Message.Large)]
+        public Message Input { get; set; }
+
+        [Params(Protocol.MsgPack, Protocol.Json)]
+        public Protocol HubProtocol { get; set; }
+
+        [GlobalSetup]
+        public void GlobalSetup()
+        {
+            switch (HubProtocol)
+            {
+                case Protocol.MsgPack:
+                    _hubProtocolReaderWriter = new HubProtocolReaderWriter(new MessagePackHubProtocol(), new PassThroughEncoder());
+                    break;
+                case Protocol.Json:
+                    _hubProtocolReaderWriter = new HubProtocolReaderWriter(new JsonHubProtocol(), new PassThroughEncoder());
+                    break;
+            }
+
+            switch (Input)
+            {
+                case Message.Small:
+                    _hubMessage = new CancelInvocationMessage("123");
```

Comment:
Let's use the same message type for all of these tests. If that makes "Small" bigger, so be it.